### PR TITLE
Move VAT exemption to UserCompanyDTO

### PR DESCRIPTION
**Pull request by malteo**

### CODE REVIEW

File: cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserCompanyApi.java [hunk 1]
```diff
@@ -14,6 +14,13 @@
...
 {
+    @POST( "userCompany" )
+    Call<UserCompanyDTO> create( @Body UserCompanyDTO company );
+
+    /**
+     * @deprecated by {@link #create(UserCompanyDTO)}
+     */
+    @Deprecated
     @POST( "userCompany" )
     Call<UserCompanyDTO> create( @Body MinimalCompanyDTO company );
```
Comment (AndreaVecchi) [inline cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/UserCompanyApi.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/UserCompanyDTO.java [hunk 2]
```diff
@@ -1,10 +1,22 @@
 package com.cloudesire.platform.apiclient.dto.model.dto;
 
 import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+import java.util.Date;
 
 @ApiModel( description = "Company data of a business customer" )
 public class UserCompanyDTO extends CompanyDTO
```
Comment (AndreaVecchi) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/UserCompanyDTO.java]: MINOR Override the "equals" method in this class. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS2160)

**Comment by AndreaVecchi** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/UserCompanyDTO.java [hunk 3]
```diff
@@ -1,10 +1,22 @@
...
 {
+    @ApiModelProperty( "The reference number of a VAT Exemption document valid under Italian laws" )
+    private String vatExemptionDeclarationNumber;
+
+    @ApiModelProperty( "The declaration date of a VAT Exemption document valid under Italian laws" )
+    private Date vatExemptionDeclarationDate;
+
+    @ApiModelProperty( "The internal protocol number of a VAT Exemption document valid under Italian laws" )
+    private String vatExemptionInternalProtocolNumber;
```
Comment (gionn) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/UserCompanyDTO.java]: move inside a new class

**Pull request merged**